File: 2418.Sort-the-People.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function (names, heights) {
  const people = [];

  for (let i = 0; i < names.length; i++) {
    people.push({ name: names[i], height: heights[i] });
  }

  people.sort((p1, p2) => p2.height - p1.height);

  return people.map((p) => p.name);
};

File: python/2418.Sort-the-People.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        height_map = {}
        sorted_names = []

        for i in range(len(names)):
            height_map[heights[i]] = names[i]

        for height, name in sorted(height_map.items(), reverse=True):
            sorted_names.append(name)

        return sorted_names